rimage/transform/parallel_projection.go

```diff
@@ -95,5 +97,162 @@ func (pp *ParallelProjection) PointCloudToRGBD(cloud pointcloud.PointCloud) (*ri
 
 // ImagePointTo3DPoint takes the 2D pixel point and assumes that it represents the X,Y coordinate in mm as well.
 func (pp *ParallelProjection) ImagePointTo3DPoint(pt image.Point, d rimage.Depth) (r3.Vector, error) {
-	return r3.Vector{float64(pt.X), float64(pt.Y), float64(d)}, nil
+	return r3.Vector{X: float64(pt.X), Y: float64(pt.Y), Z: float64(d)}, nil
+}
+
+// ParallelProjectionOntoXZWithRobotMarker allows the creation of a 2D projection of a pointcloud and robot
+// position onto the XZ plane.
+type ParallelProjectionOntoXZWithRobotMarker struct {
+	robotPose *spatialmath.Pose
+}
+
+const (
+	sigmaLevel       = 7
+	imageHeight      = 1080
+	imageWidth       = 1070
+	missThreshold    = 0.44
+	hitThreshold     = 0.53
+	voxelRadius      = 1
+	robotMarkerRatio = 5
+)
+
+// PointCloudToRGBD creates an image of a pointcloud in the XZ plane, scaling the points to a standard image
+// size. It will also add a red marker to the map to represent the location of the robot. The returned depthMap
+// is unused and so will always be nil.
+func (ppRM *ParallelProjectionOntoXZWithRobotMarker) PointCloudToRGBD(cloud pointcloud.PointCloud,
+) (*rimage.Image, *rimage.DepthMap, error) {
+	meta := cloud.MetaData()
+
+	var err error
+	var X, Z []float64
```

Comment:
Do these need to be public? If not, could we maybe use `x,z`?

Reply:
Also, would anonymous structs, like `struct{x: float64, z: float64}` rather than float64 slices, maybe make it easier to keep track of which value corresponds to x & which corresponds to z?

Reply:
Never mind, I see that the api of all the math functions we use, need the points to be in slices.

Reply:
They are not public/exposed since they only exist internal to the function. I prefer having capital letters for these because the r3.Vector value is also X and as x is often used as a dumpy variable I wanted to give it some extra weight. 

However, I dont necessarily feel all that strongly about it so can change it if you want.

Reply:
My understanding is that go uses capitol letters to denote public entities. If X & Y are not public outside of the package then I don't think it makes sense for them to be capitalized.

Also, the name of these variables made me think that they were not aggregates until I re read the type.

Could I recommend a name like `xs` `zs` or something to make it clear that we are talking about `all the x dimensions in the point cloud` and `all the z dimensions in the point cloud`?